### PR TITLE
test: rewrite finance.spec.ts

### PR DESCRIPTION
**Pull request by togami2864**
ref #287 
just rewrite `finance.spec.ts`

### CODE REVIEW

**Review by ST-DDT — DISMISSED** [review]
Looks good to me.

File: test/finance.spec.ts [hunk 1]
```diff
@@ -1,151 +1,160 @@
...
+      describe('account(length)', () => {
+        it('should supply a default length', () => {
+          const accountNum = faker.finance.account();
+          expect(accountNum).toBeTruthy();
+          expect(
+            accountNum.length,
+            'The length of the account number should be 8 characters long'
+          ).toBe(8);
+        });
+        it('should supply a default length if a zero is passed', () => {
```
Comment (Shinigami92) [inline test/finance.spec.ts:130]: **nitpick**: I prefer to have a blank line between such `it`-blocks

File: test/finance.spec.ts [hunk 2]
```diff
@@ -1,151 +1,160 @@
...
+  describe(`random seeded tests for seed ${faker.seedValue}`, () => {
+    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
+      describe('account(length)', () => {
+        it('should supply a default length', () => {
+          const accountNum = faker.finance.account();
+          expect(accountNum).toBeTruthy();
+          expect(
+            accountNum.length,
+            'The length of the account number should be 8 characters long'
+          ).toBe(8);
```
Comment (Shinigami92) [inline test/finance.spec.ts:128]: **suggestion:** There is a `toHaveLength` function that can be used here and in some other places

File: test/finance.spec.ts [hunk 3]
```diff
@@ -1,151 +1,160 @@
...
-      const parens = !ellipsis;
-
-      const mask = faker.finance.mask(length, ellipsis, parens);
-      expect(mask).toBeTruthy();
-    });
+  describe('accountName()', () => {
+    const accountName = faker.finance.accountName();
+    const expectedType = 'string';
+    expect(accountName).toBeTruthy();
+    expect(typeof accountName).toBe(expectedType);
```
Comment (Shinigami92) [inline test/finance.spec.ts:154]: **nitpick:** `expectedType` is not reused, just inline the `'string'`

Applies to other locations too

File: test/finance.spec.ts [hunk 4]
```diff
@@ -1,151 +1,160 @@
...
   });
 
-  describe('amount(min, max, dec, symbol)', () => {
+  describe('amount(min, max, dec, symbol, autoFormat)', () => {
     it('should use the default amounts when not passing arguments', () => {
       const amount = faker.finance.amount();
```
Comment (Shinigami92) [inline test/finance.spec.ts:159]: **nitpick:** Reads weird when `amount()` is used, but in the `describe` are listed all params

File: test/finance.spec.ts [hunk 5]
```diff
@@ -238,54 +247,51 @@ describe('finance', () => {
...
-  describe('ethereumAddress()', () => {
-    it('returns a random ethereum address', () => {
-      const ethereumAddress = faker.finance.ethereumAddress();
-      expect(ethereumAddress).match(/^(0x)[0-9a-f]{40}$/);
-    });
+  describe('creditCardCVV', () => {
+    const cvv = faker.finance.creditCardCVV();
+    const expectedType = 'string';
+    expect(typeof cvv).toBe(expectedType);
+    expect(cvv).match(/^[0-9]{3}$/);
```
Comment (Shinigami92) [inline test/finance.spec.ts:284]: **suggestion:** Maybe we can use `\d` here, can we?

**Review by Shinigami92 — DISMISSED** [review]

File: test/finance.spec.ts [hunk 6]
```diff
@@ -238,58 +248,52 @@ describe('finance', () => {
...
+  describe('creditCardNumber', () => {
     it('returns a random credit card number', () => {
       let number = faker.finance.creditCardNumber();
-      number = number.replace(/\D/g, ''); // remove formating
+      number = number.replace(/\D/g, ''); // remove formatting
       console.log('version:', process.version, number, number.length);
       expect(number.length).greaterThanOrEqual(13);
       expect(number.length).lessThanOrEqual(20);
-      expect(number).match(/^[0-9]{13,20}$/);
+      expect(number).match(/\d{13,20}$/);
```
Comment (ST-DDT) [inline test/finance.spec.ts:296]:
```suggestion
      expect(number).match(/^\d{13,20}$/);
```

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**